### DISCUSSION TITLE
Store and restore different pip crop's

### DISCUSSION THREAD

**Discussion by m98bw** [WindowTop/WindowTop-App#353] [Ideas]
I was wondering if you could add an option to store (and restore) crop settings, perhaps some option similar to the system used by a very similar app called ontopreplica. (how i found this project as a direct replacement) its the only part of the program ive found lacking. and one that would drastically speed up my workflows and setup times
(i pip 8 windows, then crop them, then place them, then enable transperancy and click through, the longest part of this is the crop, as i know the crop for each window but cant store that information so this forces me to re-crop the window every time). 

![image](https://github.com/WindowTop/WindowTop-App/assets/136832482/81d1efae-3779-4a43-8c15-1a3d9b30c79a)
![image](https://github.com/WindowTop/WindowTop-App/assets/136832482/b64a46f7-0ac4-444d-acb9-e33fe68251d0)
![image](https://github.com/WindowTop/WindowTop-App/assets/136832482/a5900e9a-785e-43bc-84cf-5babfe789263)

**Comment by gileli121** [comment]
@m98bw 
Thank you for the suggestion.
Your idea is to solve a problem that takes some effort to select the area to crop each time.
I think that the idea proposed here: https://github.com/WindowTop/WindowTop-App/issues/324
seems better if you ask me. What do you think?

**Reply by m98bw** [reply]
yeah that's honestly pretty close to perfect!

**Reply by gileli121** [reply]
Great, do you also prefer the `Select-element approach` that suggested in #324 ?

**Reply by m98bw** [reply]
i think its a good idea! but I'm unsure of how it would work with windows that don't have a defined space. eg Sharex uses a open source tool to find "regions" and "elements" inside windows like how Firefox and Chrome do. but its limited by what the window provides. and many windows simply don't provide any information. 

so i think select element would work well. but you'd need something like the current selection method as a fallback. 

but yes. in both cases. being able to save and load a defined area would be very useful and speed up the setup times.

**Reply by gileli121** [reply]
Indeed, the feature for saving regions likely still needs to be developed. I wasn't aware that ShareX had implemented a functionality for selecting elements in screenshots. This will be useful for inspiration.

**Reply by wooshaq** [reply]
So, any info on that?

**Reply by gileli121** [reply]
Sorry, I don't understand. Info about what? About the development effort for this feature?
I took a break in development efforts.
I will continue with development efforts in the future...
The PiP and Crop feature is very complicated. This function (crop and mouse interaction, mouse cloning, and more) took many months to complete.

There are too many good ideas. Your input is crucial. Please vote for suggestions so it will be easier to know what feature to develop next.
